plugins/koterm.koplugin/terminputtext.lua

```diff
@@ -0,0 +1,391 @@
+    maxc = 80,
+    strike_callback = nil,
+}
+
+function TermInputText:onTapTextBox(arg, ges)
+    -- disable positioning cursor by tap in emulator mode
+    if self.terminal_mode then
+        return true
+    end
+    return InputText.onTapTextBox(self, arg, ges)
+end
+
+function TermInputText:addChars(chars, skip_callback)
+    if not chars then
+        -- VirtualKeyboard:addChar(key) gave us 'nil' once (?!)
+        -- which would crash table.concat()
+        return
+    end
+    if self.enter_callback and chars == "\n" and not skip_callback then
+        UIManager:scheduleIn(0.3, function() self.enter_callback() end)
+        return
+    end
+    if self.strike_callback and not skip_callback then
+        self.strike_callback(chars)
+        return
+    end
+    if self.readonly or not self:isTextEditable(true) then
+        return
+    end
+
+    self.is_text_edited = true
+    if #self.charlist == 0 then -- widget text is empty or a hint text is displayed
+        self.charpos = 1 -- move cursor to the first position
+    end
+
+    if self.terminal_mode then
+        local chars_list = util.splitToChars(chars) -- for UTF8
+        for i = 1, #chars_list do
+            if chars_list[i] == "\n" then
+                local pos = self.charpos
+
+                -- detect current column
+                while pos > 0 and self.charlist[pos] ~= "\n" do
+                    pos = pos - 1
+                end
+                local column = self.charpos - pos
+
+                -- go to EOL
+                pos = self.charpos
+                while self.charlist[pos] and self.charlist[pos] ~= "\n" do
+                    pos = pos + 1
+                end
+
+                if not self.charlist[pos] then -- add new line if necessary
+                    table.insert(self.charlist, pos, "\n")
+                    pos = pos + 1
+                end
+
+                -- go to column in next line
+                for j = 1, column-1 do
+                    if self.charlist[pos+i] or self.charlist[pos+i] ~= "\n" then
+                        table.insert(self.charlist, pos, " ")
+                    else
+                        break
+                    end
+                end
+                self.charpos = pos
+            else -- ~="\n"
+                if self.charlist[self.charpos] == "\n" then
+                    self.charpos = self.charpos + 1
+                end
+                table.remove(self.charlist, self.charpos)
+                table.insert(self.charlist, self.charpos, chars_list[i])
+                self.charpos = self.charpos + 1
+            end
+        end
+        self:initTextBox(table.concat(self.charlist), true)
+        return
+    end
+
+    InputText.addChars(self, chars)
+end
+dbg:guard(TermInputText, "addChars",
+    function(self, chars)
+        assert(type(chars) == "string",
+            "Wrong chars value type (expected string)!")
+    end)
+
+function TermInputText:enterAlternateKeypad(maxr, maxc)
```

Comment:
May be mark a comment after your specific methods - or after the one overriden, so also we can distriguish them better.
```
-- TermInputText specific
or
-- overridden InputText method
```